src/main.rs

```diff
@@ -39,6 +39,17 @@ fn get_user_agent() -> &'static str {
 
 #[tokio::main]
 async fn main() -> Result<()> {
+    std::process::exit(inner_main().await?);
```

Comment:
Seems we would have to live with this until Rust can support [returning an exit code from `main()`](https://stackoverflow.com/a/54174631/5915221). Or we can we could use this [crate](https://docs.rs/quit/1.1.2/quit/) instead.

Reply:
`quit` looks very nice. Maybe something for after #41? I don't know if it composes with `tokio::main`.

Another option might be to attach an exit code as context to the `anyhow` error. That would work better in cases where we also want to show a message.

Reply:
if it could be done using `anyhow` then that is good. Otherwise, we could wait for https://github.com/ducaale/xh/pull/41 👍

Reply:
I looked at it again and I think the current approach is better after all. `--check-status` doesn't always print a message, and the program has to continue even after the exit code is known. So it's not like an error and you don't get much mileage out of `quit`'s stack unwinding.